lib/schemas/rfc7239.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+function splitMap (string, separator, cb) {
+  // split into elements
+  return string.split(separator)
+    .filter(Boolean)
+    .forEach(cb)
+}
+
+function parsePart (part) {
+  var pair = part.split(/ *= */)
+
+  var name = pair[0].toLowerCase()
+  var value = pair[1]
+
+  if (value) {
+    switch (typeof this[name]) {
+      case 'undefined':
+        this[name] = value
+        break
+
+      // convert to array
+      case 'string':
+        this[name] = [this[name], value]
+        break
+
+      // append to array
+      case 'object':
+        this[name].push(value)
+        break
+    }
+  }
+}
+
+var ELEMENT_SEPARATOR = / *; */
+var PART_SEPARATOR = / *, */
+
+module.exports = function (req) {
+  var forwarded = {}
+  var header = req.headers.forwarded
+
+  if (!header) {
+    return forwarded
+  }
+
+  splitMap(header, ELEMENT_SEPARATOR, function parseElement (el) {
```

Comment:
That is irrelevant; an incoming header can contain any values, valid or not. The current implementation will do the wrong thing and this is not good, as the only reason you are parsing these headers is for logging, auditing, or access controls.

Just think of a standard proxy that does not modify anything existing in the `Forwarded` header and simply adds it's own data to it; it will pass-through the bogus value that will create false parses.